src/dbwrapper.cpp

```diff
@@ -133,7 +133,7 @@ CDBWrapper::CDBWrapper(const fs::path& path, size_t nCacheSize, bool fMemory, bo
             leveldb::Status result = leveldb::DestroyDB(path.string(), options);
             dbwrapper_private::HandleError(result);
         }
-        TryCreateDirectories(path);
+        assert(TryCreateDirectories(path));
```

Comment:
`assert(…)`:s should not have side-effects :)

Reply:
Yes, sorry, I was debugging and I forgot to handle this correctly. I'll add proper error handling here in additional commit.